contracts/FundManagement.sol

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity 0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract FundManagement is Ownable {
+    struct Fund {
+        uint256 totalAmount;
+        address asset;
+        uint256 requestWithdrawTime;
```

Comment:
`requestWindow` can be changed at any point in the contract, but when you request funds you are doing them with the knowledge of the current request window; so it'd make sense for `requestWithdrawTime` to instead be `block.timestamp + requestWindow` at time of request, and for you to compare `block.timestamp` when claiming against that